background/constants/index.ts

```diff
@@ -33,6 +33,8 @@ export enum EarnStages {
   Live = "Live",
 }
 
+export const INTERNAL_PROVIDER_DOMAIN_NAME = "0x Router"
```

Comment:
hmmm I don't really like this const living here alone 😅 ideally I think we could include all internal providers' names in one object like
```suggestion
export const INTERNAL_PROVIDER_DOMAIN_NAME = {
  0X: "0x Router",
  UNISWAP: "🦄 Uniswap"
  // ...
}
```

Reply:
Small terminology point (which is reflected in the new  variable name!)--a provider for the extension specifically refers to the classes we use to connect to on chain data see `SerialFallbackProvider` and the ethers providers).

Strong support for extracting to named constants, btw, though keep in mind the contract info is already in a named constant! Waiting for Vlad's take on how this should work, but my guess is I'm going to suggest a different way of approaching this that does not involve checking string constant equality.